packages/mask/src/plugins/CryptoartAI/SNSAdaptor/ActionBar.tsx

```diff
@@ -0,0 +1,87 @@
+import { Box } from '@mui/material'
+import { makeStyles } from '@masknet/theme'
+import { useI18N } from '../../../utils'
+import { CollectibleState } from '../hooks/useCollectibleState'
+import ActionButton from '../../../extension/options-page/DashboardComponents/ActionButton'
+import { useControlledDialog } from '../../../utils/hooks/useControlledDialog'
+import { MakeOfferDialog } from './MakeOfferDialog'
+import { CheckoutDialog } from './CheckoutDialog'
+
+const useStyles = makeStyles()((theme) => {
+    return {
+        root: {
+            marginLeft: theme.spacing(-0.5),
+            marginRight: theme.spacing(-0.5),
+        },
+        button: {
+            flex: 1,
+            margin: `0 ${theme.spacing(0.5)}`,
+        },
+    }
+})
+
+export interface ActionBarProps {}
+
+export function ActionBar(props: ActionBarProps) {
+    const { t } = useI18N()
+    const { classes } = useStyles()
+    const { asset } = CollectibleState.useContainer()
+
+    const {
+        open: openCheckoutDialog,
+        onOpen: onOpenCheckoutDialog,
+        onClose: onCloseCheckoutDialog,
+    } = useControlledDialog()
+    const { open: openOfferDialog, onOpen: onOpenOfferDialog, onClose: onCloseOfferDialog } = useControlledDialog()
+
+    if (!asset.value) return null
+
+    return (
+        <Box className={classes.root} sx={{ marginTop: 1 }} display="flex" justifyContent="center">
+            {!asset.value.isSoldOut &&
+            !asset.value.is_owner &&
+            asset.value.is24Auction &&
+            new Date(asset.value.latestBidVo?.auctionEndTime).getTime() - Date.now() > 0 ? (
```

Comment:
Fallback `asset.value.latestBidVo?.auctionEndTime` to `0`, or `new Date(undefined)` will be `Invalid Date`

`asset.value` is used many times, better to assign it to a variable.

```suggestion
            new Date(assetValue.latestBidVo?.auctionEndTime ?? 0).getTime() > Date.now() ? (
```
OR

```suggestion
            assetValue.latestBidVo && new Date(assetValue.latestBidVo.auctionEndTime).getTime() > Date.now() ? (
```